cmake/gen_version.cmake

```diff
@@ -0,0 +1,59 @@
+#
+#
+#  Copyright 2016 CUBRID Corporation
+# 
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+# 
+#       http://www.apache.org/licenses/LICENSE-2.0
+# 
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+# 
+#
+find_package(Git)
+
+file(STRINGS ${CUBRID_JDBC_SOURCE_DIR}/VERSION CUBRID_JDBC_VERSION)
+
+if(NOT CUBRID_JDBC_VERSION)
+    message(FATAL_ERROR "Could not find VERSION file")
+endif(NOT CUBRID_JDBC_VERSION)
+
+set(CUBRID_JDBC_START_SERIAL_DATE "2021-03-30")
+
+if (UNIX)
+execute_process(
+    COMMAND ${GIT_EXECUTABLE} rev-list --count --after ${CUBRID_JDBC_START_SERIAL_DATE} HEAD
+    COMMAND awk "{ printf \"%04d\", $1 }"
+    OUTPUT_VARIABLE EXTRA_VERSION RESULT_VARIABLE git_result
+    ERROR_QUIET OUTPUT_STRIP_TRAILING_WHITESPACE
+    WORKING_DIRECTORY ${CUBRID_JDBC_SOURCE_DIR})
+else (UNIX)
+execute_process(
+	COMMAND ${GIT_EXECUTABLE} rev-list --count --after ${CUBRID_JDBC_START_SERIAL_DATE} HEAD
+    OUTPUT_VARIABLE EXTRA_VERSION RESULT_VARIABLE git_result
+    ERROR_QUIET OUTPUT_STRIP_TRAILING_WHITESPACE
+    WORKING_DIRECTORY ${CUBRID_JDBC_SOURCE_DIR})
+
+# getting around awk is not supported on windows
+string(CONCAT EXTRA_VERSION "0000" ${EXTRA_VERSION})
+string(REGEX MATCHALL "([0-9][0-9][0-9][0-9]$)" VERSION_MATCHES ${EXTRA_VERSION})
+  list(GET VERSION_MATCHES 0 EXTRA_VERSION)
+endif (UNIX)
+
+if(git_result)
+    message(FATAL_ERROR "Could not get count information from Git")
+endif(git_result)
+
+set(CUBRID_JDBC_RELEASE_VERSION ${CUBRID_JDBC_VERSION}.${EXTRA_VERSION})
+
+file(WRITE ${CUBRID_JDBC_OUTPUT_DIR}/VERSION-DIST "${CUBRID_JDBC_RELEASE_VERSION}")
+file(TOUCH ${CUBRID_JDBC_OUTPUT_DIR}/CUBRID-JDBC-${CUBRID_JDBC_RELEASE_VERSION})
```

Comment:
I use cmake2.8.12.2, and got error message during test build using 'build.sh'
```
-- Build files have been written to: /home/kshanrel/jc/output
Scanning dependencies of target jdbc_build
Cleaning CUBRID JDBC project and Generating JDBC VERSION-DIST file ...
-- Found Git: /usr/bin/git (found version "2.31.1")
CMake Error at /home/kshanrel/jc/cmake/gen_version.cmake:55 (file):
  file does not recognize sub-command TOUCH
```

Reply:
Thank you for your review. I found the `TOUCH` sub-command is newly added from CMake 3. It can substitute `file(WRITE...` for `file(TOUCH...` on the version 2.x. I'm going to fix it.